src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/AssemblyBuilderImpl.cs

```diff
@@ -0,0 +1,136 @@
+            var peBlob = new BlobBuilder();
+            BlobContentId contentId = peBuilder.Serialize(peBlob);
+            peBlob.WriteContentTo(peStream);
+        }
+
+        public void Save(Stream stream)
+        {
+            if (_previouslySaved) // Cannot save an assembly multiple times. This is consistent with Save() in .Net Framework.
+            {
+                throw new InvalidOperationException(SR.CannotSaveMultipleTimes);
+            }
+
+            ArgumentNullException.ThrowIfNull(stream);
+
+            if (_assemblyName == null || _assemblyName.Name == null)
+            {
+                throw new InvalidOperationException();
+            }
+
+            if (_module == null)
+            {
+                throw new InvalidOperationException(SR.AModuleRequired);
+            }
+
+            // Add assembly metadata
+            var metadata = new MetadataBuilder();
+            metadata.AddAssembly( // Metadata is added for the new assembly - Current design - metadata generated only when Save method is called.
+               metadata.GetOrAddString(value: _assemblyName.Name),
+               version: _assemblyName.Version ?? new Version(0, 0, 0, 0),
+               culture: _assemblyName.CultureName == null ? default : metadata.GetOrAddString(value: _assemblyName.CultureName),
+               publicKey: _assemblyName.GetPublicKey() is byte[] publicKey ? metadata.GetOrAddBlob(value: publicKey) : default,
+               flags: (AssemblyFlags)_assemblyName.Flags,
+               hashAlgorithm: AssemblyHashAlgorithm.None); // AssemblyName.HashAlgorithm is obsolete so default value used.
+
+            // Add module's metadata
+            _module.AppendMetadata(metadata);
+
+            var ilBuilder = new BlobBuilder();
+            WritePEImage(stream, metadata, ilBuilder);
+            _previouslySaved = true;
+        }
+
+        public void Save(string assemblyFileName)
+        {
+            ArgumentNullException.ThrowIfNull(assemblyFileName);
+
+            using var peStream = new FileStream(assemblyFileName, FileMode.OpenOrCreate, FileAccess.ReadWrite);
+            Save(peStream);
+        }
+
+        protected override ModuleBuilder DefineDynamicModuleCore(string name)
+        {
+            ArgumentException.ThrowIfNullOrEmpty(name);
+
+            if (_module == null)
+            {
+                throw new InvalidOperationException(SR.ModuleNotFound);
+            }
+
+            return _module;
+        }
+
+        protected override ModuleBuilder? GetDynamicModuleCore(string name)
+        {
+            ArgumentException.ThrowIfNullOrEmpty(name);
+
+            if (ModuleBuilderImpl.ManifestModuleName.Equals(name))
+            {
+                return _module;
+            }
+
+            return null;
+        }
+
+        protected override void SetCustomAttributeCore(ConstructorInfo con, byte[] binaryAttribute) => throw new NotSupportedException();
```

Comment:
It may be more appropriate for the not-yet-implemented methods to throw NotImplementedException. It will make it easier to search for what remains to be implemented.